pkg/cloud-provider/cloudapi/aws/aws_crd_helpers.go

```diff
@@ -76,3 +78,30 @@ func ec2InstanceToVirtualMachineCRD(instance *ec2.Instance, namespace string, ac
 		strings.ToLower(cloudNetwork), cloudNetwork, v1alpha1.VMState(*instance.State.Name), tags, networkInterfaces,
 		providerType, accountId)
 }
+
+// ec2VpcToInternalVpcObject converts ec2 vpc object to vpc runtime object.
+func ec2VpcToInternalVpcObject(vpc *ec2.Vpc, namespace string, accountName string, region string) *runtimev1alpha1.Vpc {
+	cloudName := ""
+	tags := make(map[string]string, 0)
+	if len(vpc.Tags) != 0 {
+		for _, tag := range vpc.Tags {
+			tags[*(tag.Key)] = *(tag.Value)
+		}
+		if value, found := tags["Name"]; found {
```

Comment:
should we keep it simple just like virtual machine and get rid of L90-92?
cloudName := tags[ResourceNameTagKey]

```suggestion
		if value, found := tags[ResourceNameTagKey]; found {
```

Reply:
All case related comments are taken care along with this.